samcli/lib/deploy/deployer.py

```diff
@@ -0,0 +1,383 @@
+"""
+Cloudformation deploy class which also streams events and changeset information
+"""
+
+# Copyright 2012-2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from collections import OrderedDict
+import logging
+import time
+from datetime import datetime
+
+import botocore
+import click
+import pytz
+
+from samcli.commands.deploy.exceptions import DeployFailedError, ChangeSetError, DeployStackOutPutFailedError
+from samcli.commands._utils.table_print import pprint_column_names, pprint_columns
+from samcli.commands.deploy import exceptions as deploy_exceptions
+from samcli.lib.package.artifact_exporter import mktempfile, parse_s3_url
+
+LOG = logging.getLogger(__name__)
+
+DESCRIBE_STACK_EVENTS_FORMAT_STRING = (
+    "{ResourceStatus:<{0}} {ResourceType:<{1}} {LogicalResourceId:<{2}} {ResourceStatusReason:<{3}}"
+)
+DESCRIBE_STACK_EVENTS_DEFAULT_ARGS = OrderedDict(
+    {
+        "ResourceStatus": "ResourceStatus",
+        "ResourceType": "ResourceType",
+        "LogicalResourceId": "LogicalResourceId",
+        "ResourceStatusReason": "ResourceStatusReason",
+    }
+)
+
+DESCRIBE_CHANGESET_FORMAT_STRING = "{Operation:<{0}} {LogicalResourceId:<{1}} {ResourceType:<{2}}"
+DESCRIBE_CHANGESET_DEFAULT_ARGS = OrderedDict(
+    {"Operation": "Operation", "LogicalResourceId": "LogicalResourceId", "ResourceType": "ResourceType"}
+)
+
+OUTPUTS_FORMAT_STRING = "{OutputKey:<{0}} {OutputValue:<{1}} {Description:<{2}}"
+OUTPUTS_DEFAULTS_ARGS = OrderedDict(
+    {"OutputKey": "OutputKey", "OutputValue": "OutputValue", "Description": "Description"}
+)
+
+
+class Deployer:
+    def __init__(self, cloudformation_client, changeset_prefix="samcli-cloudformation-package-deploy-"):
+        self._client = cloudformation_client
+        self.changeset_prefix = changeset_prefix
+        self.client_sleep = 0.5
```

Comment:
what is this used for? what is the unit?

Reply:
added a comment